clang/lib/CIR/Dialect/Transforms/LibOpt.cpp

```diff
@@ -120,47 +120,37 @@ static bool containerHasStaticSize(StructType t, unsigned &size) {
 }
 
 void LibOptPass::xformStdFindIntoMemchr(StdFindOp findOp) {
-  // First and second operands need to be iterators begin() and end().
-  // TODO: look over cir.loads until we have a mem2reg + other passes
-  // to help out here.
-  auto iterBegin = dyn_cast<IterBeginOp>(findOp.getOperand(0).getDefiningOp());
-  if (!iterBegin)
-    return;
-  if (!isa<IterEndOp>(findOp.getOperand(1).getDefiningOp()))
-    return;
-
-  // Both operands have the same type, use iterBegin.
-
-  // Look at this pointer to retrieve container information.
-  auto thisPtr =
-      iterBegin.getOperand().getType().cast<PointerType>().getPointee();
-  auto containerTy = dyn_cast<StructType>(thisPtr);
-  if (!containerTy)
-    return;
-
-  if (!isSequentialContainer(containerTy))
```

Comment:
Instead of deleting this call, it would be best if this renamed to `isSequentialContainerOrRawPointers` and embed the raw pointer check logic inside it.

Reply:
I'm not sure what you're looking for here. Could you elaborate a bit?

Reply:
I somehow missed those calls later in the diff!